facade/service.go

```diff
@@ -64,17 +65,19 @@ var (
 
 // AddService adds a service; return error if service already exists
 func (f *Facade) AddService(ctx datastore.Context, svc service.Service) (err error) {
-	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.AddService"))
 	var tenantID string
+	defer ctx.Metrics().Stop(ctx.Metrics().Start("Facade.AddService"))
+	alog := f.auditLogger.Action("Add").Message(ctx, "Adding new Service: " + svc.Name).ID(svc.GetID()).Type(svc.GetType())
```

Comment:
This particular method is a case where the defer would work really well. Instead of modifying lines 75 and 80, you could simply add `defer alog.Error(err)` at line 71
then at the bottom, 
```
     err = f.addService(ctx, tenantID, svc, false)
     return err
```